internal/ingress/controller/template/template.go

```diff
@@ -1672,3 +1673,42 @@ func convertGoSliceIntoLuaTable(goSliceInterface interface{}, emptyStringAsNil b
 		return "", fmt.Errorf("could not process type: %s", kind)
 	}
 }
+
+// buildOriginRegex returns an origin as a regex
+func buildOriginRegex(origin string) string {
+	if !strings.HasPrefix(origin, "*") {
+		return origin
+	}
+
+	origin = strings.Replace(origin, "*.", "", 1)
+	parts := strings.Split(origin, ".")
+
+	return "(" + strings.Join(parts, "\\.") + ")"
```

Comment:
Use `https://pkg.go.dev/regexp#QuoteMeta` to escape the regex, don't try and escape it yourself.